nvflare/app_common/workflows/hierarchical_statistics_controller.py

```diff
@@ -0,0 +1,335 @@
+                self.system_panic(
+                    f"Failed to decode hierarchy config JSON from the file {hierarchy_config_file_path}: {e}", fl_ctx
+                )
+                return False
+            except Exception as e:
+                self.system_panic(
+                    f"An unexpected error occurred while loading hierarchy config file {hierarchy_config_file_path}: {e}",
+                    fl_ctx,
+                )
+                return False
+
+        self.log_info(fl_ctx, f"start prepare inputs for task {statistic_task}")
+        inputs = self._prepare_inputs(statistic_task)
+        results_cb_fn = self._get_result_cb(statistic_task)
+
+        self.log_info(fl_ctx, f"task: {self.task_name} statistics_flow for {statistic_task} started.")
+
+        if abort_signal.triggered:
+            return False
+
+        task_props = {StC.STATISTICS_TASK_KEY: statistic_task}
+        task = Task(name=self.task_name, data=inputs, result_received_cb=results_cb_fn, props=task_props)
+
+        self.broadcast_and_wait(
+            task=task,
+            targets=None,
+            min_responses=self.min_clients,
+            fl_ctx=fl_ctx,
+            wait_time_after_min_received=self.wait_time_after_min_received,
+            abort_signal=abort_signal,
+        )
+
+        self.global_statistics = get_global_stats(
+            self.global_statistics, self.client_statistics, statistic_task, hierarchy_config_json
+        )
+
+        self.log_info(fl_ctx, f"task {self.task_name} statistics_flow for {statistic_task} flow end.")
+
+    def _recursively_round_global_stats(self, global_stats):
+        """Apply given precision to the calculated global statistics.
+
+        Args:
+            global_stats: Global stats.
+
+        Returns:
+            A dict containing global stats with applied precision.
+        """
+        if isinstance(global_stats, dict):
+            for key, value in global_stats.items():
+                if key == StC.GLOBAL or key == StC.LOCAL:
+                    for key, metric in value.items():
+                        if key == StC.STATS_HISTOGRAM:
+                            for ds in metric:
+                                for name, val in metric[ds].items():
+                                    hist: Histogram = metric[ds][name]
+                                    buckets = StatisticsController._apply_histogram_precision(hist.bins, self.precision)
+                                    metric[ds][name] = buckets
+                        else:
+                            for ds in metric:
+                                for name, val in metric[ds].items():
+                                    metric[ds][name] = round(metric[ds][name], self.precision)
+                    continue
+                if isinstance(value, list):
+                    for item in value:
+                        self._recursively_round_global_stats(item)
+        if isinstance(global_stats, list):
```

Comment:
Change to "elif".

Reply:
Sounds good, done, thanks for pointing it out.